Recast/PlayerViewController.swift

```diff
@@ -8,59 +8,344 @@
+        controlsView.rewindButton.addTarget(self, action: #selector(skipBackButtonWasPressed(_:)), for: .touchUpInside)
+        controlsView.timeSlider.addTarget(self, action: #selector(timeSliderDidChange(_:)), for: .valueChanged)
 
-        setUpConstraints()
+        setupConstraints()
     }
 
-    func setUpConstraints() {
+    func setupConstraints() {
         // MARK: - Constants
         let topPadding: CGFloat = 100
-        let controlsHeight: CGFloat = 100
 
-        playerHeaderView.snp.makeConstraints { make in
-            make.left.right.top.equalToSuperview()
-            make.height.equalTo(topPadding)
+        playerView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+
+        episodeImageView.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview().inset(24)
+            make.centerY.equalToSuperview()
+            make.height.equalTo(episodeImageView.snp.width)
+        }
+
+        containerView.snp.makeConstraints { make in
+            make.leading.trailing.top.equalToSuperview()
+            make.bottom.equalTo(controlsView.snp.top)
         }
 
         controlsView.snp.makeConstraints { make in
-            make.left.right.equalToSuperview()
+            make.leading.trailing.equalToSuperview()
             make.bottom.equalToSuperview().inset(topPadding)
-            make.height.equalTo(controlsHeight)
         }
     }
 
-    @objc func collapse() {
-        self.dismiss(animated: true, completion: nil)
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        /*
+         Update the UI when these player properties change.
+
+         Use the context parameter to distinguish KVO for our particular observers
+         and not those destined for a subclass that also happens to be observing
+         these properties.
+         */
+        durationObserverToken = observe(\.player.currentItem?.duration, options: [.new, .initial]) { (strongSelf, change) in
+            // Update `timeSlider` and enable / disable controls when `duration` > 0.0.
+
+            /*
+             Handle `NSNull` value for `NSKeyValueChangeNewKey`, i.e. when
+             `player.currentItem` is nil.
+             */
+            let newDuration: CMTime
+            if let t = change.newValue, let time = t {
+                newDuration = time
+            } else {
+                newDuration = CMTime.zero
+            }
+            strongSelf.updateDuration(newDuration)
+        }
+
+        rateObserverToken = observe(\.player.rate, options: [.new, .initial]) { (strongSelf, change) in
+            // Update `playPauseButton` image.
+            let newRate = change.newValue!
+            let buttonImageName = newRate == 0.0 ? "player_play_icon" : "player_pause_icon"
+            let buttonImage = UIImage(named: buttonImageName)
```

Comment:
can you use image literals here instead of `named`?

Reply:
I thought we were trying to not use image literals?